sqlmesh/core/engine_adapter/bigquery.py

```diff
@@ -886,10 +927,41 @@ def _get_data_objects(
                 schema=row.schema_name,  # type: ignore
                 name=row.name,  # type: ignore
                 type=DataObjectType.from_str(row.type),  # type: ignore
+                clustering_key=f"({row.clustering_key})" if row.clustering_key else None,  # type: ignore
             )
             for row in df.itertuples()
         ]
 
+    def _change_clustering_key(
+        self, table: exp.Table, cluster_by: t.List[exp.Expression]
+    ) -> AlterTableOperation:
+        def _do_change() -> exp.Query:
+            bq_table = self._get_table(table)
+
+            rendered_columns = [c.sql(dialect=self.dialect) for c in cluster_by]
+            bq_table.clustering_fields = rendered_columns
+
+            self._db_call(self.client.update_table, table=bq_table, fields=["clustering_fields"])
+
+            # BigQuery only applies new clustering going forward, so this rewrites the columns to apply the new clustering to historical data
+            # ref: https://cloud.google.com/bigquery/docs/creating-clustered-tables#modifying-cluster-spec
+            self.execute(exp.update(table, {c: c for c in cluster_by}, where=exp.true()))
+
+            return exp.select("1")  # dummy query to satisfy the SourceQuery interface
```

Comment:
A downside of using `SourceQuery` is that the interface requires a `Query` (SELECT) to be returned.

I'm wondering if a new `SourceCommand` interface should be introduced instead (and `SourceQuery` changed to extend `SourceCommand`)